### PR TITLE
Fail if a `<label>` contains more than one labelable item.

### LINKED ISSUE #103: Error if you put two controls / links in one <label> element
This example is invalid html5 - it causes pretty broken behavior with VoiceOver in Safari or Chrome.

We should make this an error, it's never technically correct to put multiple links or controls inside one label element.
```
<label>
  Row 1
  <input type="checkbox">
  <a href="#">More info</a>
  <button>Do something</button>
</label>
```

### PR DESCRIPTION
**Pull request by ckundo**
[Closes #103]

### CODE REVIEW

**Comment by ckundo** [comment]
@googlebot I signed it!

**Comment by ckundo** [comment]
@alice I updated this PR to look at labelable elements only. PTAL!

File: src/js/AccessibilityUtils.js [hunk 1]
```diff
@@ -32,6 +32,20 @@ axs.utils.FOCUSABLE_ELEMENTS_SELECTOR =
     '[tabindex]';
 
 /**
+ * @const
+ * @type {string}
+ */
+axs.utils.LABELABLE_ELEMENTS_SELECTOR =
```
Comment (ckundo) [inline src/js/AccessibilityUtils.js]: These come from https://html.spec.whatwg.org/multipage/forms.html#category-label

File: test/audits/multiple-labelable-elements-per-label-test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  var result = rule.run({ scope: fixture });
+  var expected = {
+    elements: [],
+    result: axs.constants.AuditResult.PASS
+  };
+
+  deepEqual(result, expected, "passes the audit with no matching elements");
+});
+
+test("multiple labelable elements within a label tag", function() {
```
Comment (alice) [inline test/audits/multiple-labelable-elements-per-label-test.js]: What happens/should happen if we do
```html
<label for="foo1 foo2">
<input id=foo1>
<input id=foo2>
```
?

Comment (alice) [inline test/audits/multiple-labelable-elements-per-label-test.js]: Hm, though I guess that's somewhat a separate issue.

File: src/audits/MultipleLabelableElementsPerLabel.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  name: "multipleLabelableElementsPerLabel",
+  heading: "A label element may not have labelable descendants other than its labeled control.",
+  url: "https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#-ax_text_03--labels-should-only-contain-one-labelable-element",
+  severity: axs.constants.Severity.SEVERE,
+  relevantElementMatcher: function(element) {
+    return axs.browserUtils.matchSelector(element, "label");
+  },
+  test: function(scope) {
+    var controls = scope.querySelectorAll(axs.utils.LABELABLE_ELEMENTS_SELECTOR);
+    if(controls.length > 1)
```
Comment (alice) [inline src/audits/MultipleLabelableElementsPerLabel.js]: nit: space between if and (

File: src/js/AccessibilityUtils.js (hunk 1 above)
Comment (alice) [inline src/js/AccessibilityUtils.js]: Mind throwing that link in the comment?

**Comment by alice** [comment]
:+1: once the nit is addressed and Changelog.md updated.

**Comment by ckundo** [comment]
@alice I'm out of date on the release flow. Would we treat this as a release? Should I create a new heading in the changelog?

**Comment by alice** [comment]
Apologies - to clarify, please add an entry to the changelog for this change. Take a look at the Changelog to see other examples.

File: src/js/AccessibilityUtils.js [hunk 4]
```diff
@@ -32,6 +32,21 @@ axs.utils.FOCUSABLE_ELEMENTS_SELECTOR =
     '[tabindex]';
 
 /**
+ * Elements that can have labels: https://html.spec.whatwg.org/multipage/forms.html#category-label
```
Comment (ckundo) [inline src/js/AccessibilityUtils.js]: Added the link here.

**Pull request merged**